spring-rabbit/src/test/java/org/springframework/amqp/rabbit/core/RabbitTemplatePublisherCallbacksIntegrationTests.java

```diff
@@ -205,26 +190,18 @@ public Void doInRabbit(Channel channel) throws Exception {
 	public void testPublisherConfirmWithSendAndReceive() throws Exception {
 		final CountDownLatch latch = new CountDownLatch(1);
 		final AtomicReference<CorrelationData> confirmCD = new AtomicReference<CorrelationData>();
-		templateWithConfirmsEnabled.setConfirmCallback(new ConfirmCallback() {
-
-			@Override
-			public void confirm(CorrelationData correlationData, boolean ack, String cause) {
-				confirmCD.set(correlationData);
-				latch.countDown();
-			}
+		templateWithConfirmsEnabled.setConfirmCallback((correlationData, ack, cause) -> {
+			confirmCD.set(correlationData);
+			latch.countDown();
 		});
 		SimpleMessageListenerContainer container = new SimpleMessageListenerContainer(this.connectionFactoryWithConfirmsEnabled);
 		container.setQueueNames(ROUTE);
-		container.setMessageListener(new MessageListenerAdapter(new Object() {
-
-			@SuppressWarnings("unused")
-			public String handleMessage(String in) {
-				return in.toUpperCase();
-			}
-		}));
+		container.setMessageListener(
+				new MessageListenerAdapter((ReplyingMessageListener<String, String>) in -> in.toUpperCase()));
```

Comment:
I've just merged this, but I think it won't hurt if we discussion it a bit more.
1. My IDEA says me that this can be simplified as method reference `String::toUpperCase`. But, eh, it is just this case and that's all.
2. How about to improve it to something like this?
```
MessageListenerAdapter messageListener = new MessageListenerAdapter();
messageListener.<String, String>setDelegate(String::toUpperCase);
```
where the target code is:
```
public <I, O> void setDelegate(ReplyingMessageListener<I, O> delegate) {
```
Eh... One more ctor with generics doesn't work.

And also I think this Lambda deserves some Docs in favor of Java 8 universe :smile: 

WDYT?